src/GraphQLParser/ParserContext.Parse.cs

```diff
@@ -855,7 +855,6 @@ private GraphQLValue ParseListValue(bool isConstant)
 
         val.Comment = GetComment();
         val.Values = ZeroOrMore(TokenKind.BRACKET_L, isConstant ? constant : value, TokenKind.BRACKET_R);
-        val.Value = _source.Slice(start, _currentToken.End - start - 1);
```

Comment:
`Value` property for both `ListValue` and `ObjectValue` was a mistake. These are composite objects that do not have their own data like any other "top-level" AST nodes. Only "primitive" AST nodes should have `ROM Value { get; set; }` property.

Reply:
I added `IHasValueNode` interface for all "primitive" nodes.